tests/Service/PromotionsService/FindActivePromotionsByProgrammeTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use DateTimeImmutable;
+
+class FindActivePromotionsByProgrammeTest extends AbstractPromotionServiceTest
+{
+    public function testServiceFindActivePromotions()
+    {
+        $mockProgramme = $this->getMockProgramme();
+        $dateTime = new DateTimeImmutable('1990-01-01 13:30:50');
+
+        $this->mockRepository
+            ->method('findActivePromotionsByPid')
+            ->with($mockProgramme->getPid(), $dateTime, 300, 0)
```

Comment:
This use of $mockProgramme has realised we should be passing a db ID into the repository function, this would allow us to save joining onto the core_entity table to find the DB id with that pid. Once that is done, $this->getMockProgramme can be replaced with `$this->getEntity('Brand', 1)`. Check FindByContributionToProgrammeTest for an example of doing this